janus-idp/backstage-showcase
# feat(OIDC): adds a forked version of the OIDC auth plugin

## Description

Forked the Auth backend OIDC provider module from Upstream into the Showcase. The purpose of this is to address the sign-out issue currently faced by users who are using OAuth2 Proxy and Keycloak. This fork will enable them to switch to the OIDC provider and Keycloak, providing the ability to log out properly.

This PR also introduces some documentation on configuring auth providers within showcase. I settled for the inclusion of a few simple providers with very little detail at this time, with the exception of the OIDC provider. With this provider, I opted to an additional example since we will be trying to recommend this over OAuth2 Proxy.

If we wish to expand upon this further, I believe we should open up a separate issue that tackles the auth providers as a whole. Just wanted to at least have something available to begin the work.

## Which issue(s) does this PR fix

- Fixes #?

## PR acceptance criteria

Please make sure that the following steps are complete:


## How to test changes / Special notes to the reviewer

The example given within the documentation should be sufficient for testing the OIDC provider with a Keycloak instance. This example can be found in `auth.md` under the section titled `OIDC`.

Also, the changes that were made include: 
- `packages/backend/package.json` where the internal plugin was added
- `packages/backend/src/modules/authProvidersModule.ts` where we are using the forked OIDC module
- `plugins/auth-backend-module-oidc-provider/package.json` pin dependencies, update the name to internal, remove start script
- `plugins/auth-backend-module-oidc-provider/src/authenticator.ts` adds logout on Line 186 
- `showcase-docs/auth.md` adds some basic authentication docs with a few providers listed as a starting point
- `showcase-docs/getting-started.md` updates the authentication section to point to the new auth doc

Everything else surrounding the forked plugin has been relatively left in tact with the exception of the removal of the changelog, api-report, etc.

EDIT:
Some more changes that were made to fix the failing CI
- `plugins/auth-backend-module-oidc-provider/authenticator.test.ts` and `module.test.ts` added comment for sonarcloud to ignore a couple of URLs in the test
- `.rhdh/docker/Dockerfile` and `docker/Dockerfile` added the plugin module to the dockerfile.
